### PR TITLE
Read exactly UInt64's size from timerfd

### PR DESCRIPTION
**Pull request by Bouke**
On 32-bit systems we would try to read 4 byte, which is incorrect. From `man timerfd_read`:
> (...) then the buffer given to read(2) returns an unsigned 8-byte integer (uint64_t) containing the number of expirations that  have  occurred (...)
> A  read(2) will fail with the error EINVAL if the size of the supplied buffer is less than 8 bytes.

See #1337.

### CODE REVIEW

File: Sources/NIO/Selector.swift [hunk 1]
```diff
@@ -549,7 +549,7 @@ final class Selector<R: Registration> {
                 // Consume event
                 var val: UInt = 0
```
Comment (weissi) [inline Sources/NIO/Selector.swift]: sorry, my bad, the `var val` needs to become `UInt64` too.

File: Sources/NIO/Selector.swift [hunk 2]
```diff
@@ -549,7 +549,7 @@ final class Selector<R: Registration> {
...
                 // We are not interested in the result
-                _ = Glibc.read(self.timerFD, &val, MemoryLayout<UInt>.size)
+                _ = Glibc.read(self.timerFD, &val, 8)
```
Comment (weissi) [inline Sources/NIO/Selector.swift]: instead of a hard-coded `8` we should use `MemoryLayout.size(ofValue: val)` here.

**Comment by weissi** [comment]
@Lukasa usually, we'd ask for a test here but this ought to be tested by lots of tests already and writing a targeted one isn't super easy. WDYT?

File: Sources/NIO/Selector.swift (hunk 2 above)
Comment (Bouke) [reply Sources/NIO/Selector.swift]: Ah yes, we're writing to `&val` indeed.

**Comment by Lukasa** [comment]
I’m happy to skip having a test for this case, we set timers all over the shop.

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Pull request merged**